pkg/network/network-policy_test.go

```diff
@@ -0,0 +1,129 @@
+package network_test
+
+import (
+	"encoding/json"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/network"
+)
+
+var _ = Describe("Render Network Policy", func() {
+	DescribeTable("should render the network policy according when running on",
+		func(info network.ClusterInfo, expected []netv1.NetworkPolicy) {
+			obj, err := network.RenderNetworkPolicy("../../data", info)
+			Expect(err).NotTo(HaveOccurred())
+			bytes, err := json.Marshal(obj)
+			Expect(err).NotTo(HaveOccurred())
+			var nps []netv1.NetworkPolicy
+			Expect(json.Unmarshal(bytes, &nps)).To(Succeed())
+			Expect(nps).To(ConsistOf(expected))
+		},
+		Entry("vanilla k8s",
+			network.ClusterInfo{OpenShift4: false},
+			testNetworkPolices("kube-system", "k8s-app", "kube-dns", 53),
+		),
+		Entry("openshift",
+			network.ClusterInfo{OpenShift4: true},
+			testNetworkPolices("openshift-dns", "dns.operator.openshift.io/daemonset-dns", "default", 5353),
+		),
+	)
+})
+
+func testNetworkPolices(dnsNamespace, dnsLabelKey, dnsLabelValue string, dnsPort int32) []netv1.NetworkPolicy {
+	return []netv1.NetworkPolicy{
+		newNetPolAllowEgressClusterDNS(dnsNamespace, dnsLabelKey, dnsLabelValue, dnsPort),
+		newNetPolAllowEgressClusterAPI(),
+		newNetPolAllowIngressMetricEndpoints(),
+	}
+}
+
+const (
+	optInLabelAllowEgressClusterServices  = "allow-access-cluster-services"
+	optInLabelAllowIngressMetricEndpoints = "allow-access-metrics-endpoints"
+)
+
+var netPolTypeMeta = metav1.TypeMeta{
+	APIVersion: "networking.k8s.io/v1",
+	Kind:       "NetworkPolicy",
+}
```

Comment:
please move to beginning of the file

Reply:
Note that these are used by the testNetworkPolices helpers and not in the actual test code

Reply:
of course, but it is still nicer to have it in the ut file itself in the beginning and not in the middle please

Reply:
Done